### PR TITLE
Refactoring DataScienceCluster API as required by #476

### PR DESCRIPTION
**Pull request by etirelli**
Refactoring DataScienceCluster API as required by #476 
Replacing the "enabled" component field by "managementState"
Updating examples and readme file

## How Has This Been Tested?
manually in the cluster

## Merge criteria:
- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/component.go [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+	operatorv1 "github.com/openshift/api/operator/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type Component struct {
-	// Set to "true" to enable component, "false" to disable component. A disabled component will not be installed.
-	Enabled bool `json:"enabled"`
+	// Set to "managed" to enable the component, and to "removed" to disable it.
```
Comment (zdtsw) [inline components/component.go:11]:
```suggestion
	// Set to "managed" to enable the component, and to "removed" to disable it.
	// +kubebuilder:validation:Enum=managed;removed
```
to set enum in list, if user input anything else wont be accepted

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-apiVersion: operators.coreos.com/v1alpha1
+apiVersion: operators.coreos.com/v1
```
Comment (zdtsw) [inline bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:1]: hum, i think we need rework on the "bundle" part.
not sure why this is updated to v1 on CSV

**Comment by zdtsw** [comment]
https://github.com/etirelli/opendatahub-operator/pull/1 for fix ci-index job.

File: .github/workflows/sync_dashboard_crds.yaml [hunk 3]
```diff
@@ -1,29 +1,41 @@
-# Opens a PR in the opendatahub-operator repo when crds are updated in the dashboard repo
-name: Sync Dashboard CRDs
+# Opens a PR in the opendatahub-operator repo to sync crds from the dashboard repo
+name: Sync Dashboard CRDs for Operator v2.x
```
Comment (VaishnaviHire) [inline .github/workflows/sync_dashboard_crds.yaml:2]: This change is already merged?  Probably need to rebase

Comment (VaishnaviHire) [reply .github/workflows/sync_dashboard_crds.yaml:2]: @etirelli

**Comment by VaishnaviHire** [comment]
/hold

@etirelli Were you able to do auto upgrade of the crd?

While testing I get this error for operator install `risk of data loss updating "dscinitializations.dscinitialization.opendatahub.io": new CRD removes version v1alpha1 that is listed as a stored version on the existing CRD` for DSCInitialization.

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**